subiquity/ui/views/filesystem/partition.py

```diff
@@ -405,6 +406,9 @@ def __init__(self, parent, disk, partition=None):
                 lvm_names.remove(partition.name)
         else:
             initial['fstype'] = 'ext4'
+            if gap:
```

Comment:
I think that if partition is None, gap cannot be None here? If not I'm getting confused

Reply:
The intention was that a gap or a partition would be supplied, but I seem to have forgot that with this check.  I have backed this out and used a more stringent up-front check.